redash/models.py

```diff
@@ -682,8 +682,8 @@ def get_latest(cls, data_source, query, max_age=0):
     def store_result(cls, org, data_source, query_hash, query, data, run_time, retrieved_at):
         try:
             data_scanned_information = json.loads(data)['data_scanned']
-        except (ValueError, TypeError) as e:
-            data_scanned_information = 'error'
+        except (ValueError, TypeError, KeyError) as e:
```

Comment:
It looks like `KeyError` should be handled in a separate except block so that malformed JSON still gets reported as an error.